railties/test/generators/plugin_generator_test.rb

```diff
@@ -148,6 +148,32 @@ def test_generating_adds_dummy_app_without_sprockets
     end
   end
 
+  def test_generator_for_yarn
```

Comment:
@y-yagi I've just implemented it.
Itl definitely help me to do more dryer tests in this PR: https://github.com/rails/rails/pull/30101. 
Also, I've have prepared new changes to dry tests in `railties/test/generators/app_generator_test.rb` and `railties/test/generators/plugin_generator_test.rb`.
When it merges, I will open new PR with this changes.

Thank you!